interfaces/hotplug/deviceinfo.go

```diff
@@ -89,3 +90,43 @@ func (h *HotplugDeviceInfo) Attribute(name string) (string, bool) {
 	val, ok := h.Data[name]
 	return val, ok
 }
+
+func (h *HotplugDeviceInfo) firstAttrValueOf(tryAttrs ...string) string {
+	for _, attr := range tryAttrs {
+		if val, ok := h.Attribute(attr); ok && val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func (h *HotplugDeviceInfo) String() string {
+	var str []string
+
+	if devname := h.DeviceName(); devname != "" {
+		str = append(str, fmt.Sprintf("devname:%s", devname))
+	} else {
+		if devpath := h.DevicePath(); devpath != "" {
+			str = append(str, fmt.Sprintf("devpath:%s", devpath))
+		}
+	}
+	for _, attr := range []string{"MAJOR", "MINOR"} {
+		if val, ok := h.Attribute(attr); ok {
+			str = append(str, fmt.Sprintf("%s:%s", strings.ToLower(attr), val))
+		}
+	}
+
+	if vendor := h.firstAttrValueOf("ID_VENDOR_FROM_DATABASE", "ID_VENDOR_ID", "ID_VENDOR"); vendor != "" {
```

Comment:
`ID_VENDOR_ID` is just the number but `ID_VENDOR` is the name, should those be swapped? If not can you justify the ordering in a comment please?

Reply:
Thanks, you're right, reversed them.